third_party/amd/lib/TritonAMDGPUToLLVM/ReduceOpToLLVM.cpp

```diff
@@ -17,6 +17,129 @@ using ::mlir::triton::gpu::getTotalElemsPerThread;
 
 namespace AMD {
 namespace {
+struct ReduceOpPromotionConversion
+    : public ConvertTritonGPUReduceScanToLLVMPattern<triton::ReduceOp> {
+public:
+  ReduceOpPromotionConversion(
+      TritonGPUToLLVMTypeConverter &typeConverter, ModuleAllocation &allocation,
+      ConvertTritonGPUOpToLLVMPatternBase::IndexCacheInfo &indexCacheInfo,
+      int computeCapability, PatternBenefit benefit)
+      : ConvertTritonGPUReduceScanToLLVMPattern<triton::ReduceOp>(
+            typeConverter, allocation, indexCacheInfo, benefit),
+        computeCapability(computeCapability) {}
+
+  LogicalResult
+  matchAndRewrite(triton::ReduceOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    
+    // promote op
+    rewriter.modifyOpInPlace(op, [&]() {
+      // promote operands
+      SmallVector<Value> newOperands;
+      for (OpOperand &operand : op->getOpOperands()) {
+        auto val = operand.get();
+        auto oldType = val.getType().cast<RankedTensorType>();
+        auto elemType = oldType.getElementType();
+        if (elemType.isInteger(16) || elemType.isInteger(8)) {
+          auto newType = oldType.cloneWith(std::nullopt, i32_ty);
+          auto promotedVal =
+              rewriter.create<mlir::arith::ExtSIOp>(op->getLoc(), newType, val);
+          newOperands.push_back(promotedVal);
+        } else if (elemType.isF16()) {
+          auto newType = oldType.cloneWith(std::nullopt, f32_ty);
+          auto promotedVal =
+              rewriter.create<mlir::arith::ExtFOp>(op->getLoc(), newType, val);
+          newOperands.push_back(promotedVal);
+        } else {
```

Comment:
Do we support f8?
And for bf16, can you double check if it is still represented as i16 in the current LLVM version used by triton? i16 is no longer used for bfloat16 in a recent [PR](https://github.com/llvm/llvm-project/pull/80908).

Reply:
F8 is not in the unit tests and I don't there is an expectation that we will support it. I looked at the ttir for a bfloat test and it loads an f32 value so the llir is also loading a float32. I think bf16 is cast to float in the fronted.